done rough draft

ayechanst opened the pull request:
Finished the rough draft. theres some spelling mistakes im about to fix, there's some confusing wording to also fix, steve and i were thinking about making the guiding questions individual toggles, and i could not reverse engineer/explain the math well so i said to put it in a box but we should explain the math, i just couldnt figure it out myself. but here are my changes!

steve0xp reviewed (changes requested):
Great work Aye Chan! I carried out a diff check between the current SE-2 DEX README and what you have created. Please create a PR to the current DEX challenge README with your changes so I can write my comments on it there officially and we can proceed to getting it closer to being merged into the official challenge itself. I would suggest bringing this up to Austin and Zak before or after making the PR, it is up to you.

Notes / Comments

1. Overall I really like what you have done. Adding this extra content, especially the “guides” helps provide more context and assistance in an educational format. The real trick is to see if it helps a random new person to Smart Contract development or not. We can tweak it after getting feedback from others (Austin and Zak) and in “prod” (aka giving it to people coming into the DEX challenge).
2. I think the wording of this question is a bit confusing for what we need.
    1. Do we want to provide liquidity if the contract already has liquidity? How can we help/prevent this from happening?
3. Within the pricing, what do you mean by “black box” the math? Did I write that? Ultimately we want them to understand the math. It is a core competency I’d say if they want to get further in DeFi. That gets me thinking though… perhaps there could be a checklist of “core competencies” that are part of each challenge. That way, if students come along and they don’t care about the math, they can ignore it. We can discuss this more in the PR.
4. I wonder if we should make the guiding questions each their own toggle. That way, the student can get one hint at a time instead of seeing all the hints. Students rob themselves of education when they see all the hints or solutions. I think that should be stressed actually at the beginning of the tutorial if it isn’t already.
5. Check the last couple of sections you’ve changed for spelling. Altogether check the whole thing for spelling, but I think I mainly just saw a couple in the last sections.

ayechanst commented:
2. I re-worded this concern
3. What I meant by black box is that I spent a lot of time trying to reverse engineer the math to ask leading questions for students to find the answer themselves. I struggle with math though and needed help finding a better explanation for it so I moved on and just said to "black box" it since the challenge can be completed without knowing the why and how. That's why the toggle says Guided Explanation instead of Guiding Questions because I don't know enough about it to ask the right questions.  But ideally I'd like to lead them through the math I just need help or someone else can write that part? I do think core competencies is a great idea.
4. I'll make each guiding question a toggle tomorrow.
5. I ran a the document through a spell checker and added commas and fixed typos

rin-st commented:
Great job @ayechanst ! I like the approach. I believe it will help new students to understand challenge better. 

Also I'm agree that every question should be in different toggle and it needs some small additional changes, but overall looking good for me.

And if we decide to add guiding questions, we also need to add it to every other challenge too.

@ZakGriffith @austintgriffith what do you think about it?

cc @carletex @technophile-04 @Pabl0cks

ayechanst commented:
Yes I agree @rin-st, I was planning on doing this for all the challenges. Just wanted to show you guys a sample to make sure it was worth doing for all of them.

ZakGriffith commented:
After the updates from the @rin-st suggestions, this is looking really good.  I am a fan of the nested guidance which should help those who *really* want to learn the DEX to try and figure it out for themselves.  The DEX readMe has always been a bit different than the others with all it's expanding solutions, so it's a great place to start with the Guided Questions.  I say we merge this and see how it plays out, then test out how the same format could apply to the other challenges.

steve0xp commented:
@ayechanst did you end up making each guiding question its own toggle? We could just wait and see how your changes so far fair with actual students. I think the changes you've made overall from what is there currently is great, so I'm game with you merging it now. @ZakGriffith I think you or Austin or Carlos or other admins would do the merge right?

rin-st commented on README.md:
```diff
@@ -67,15 +67,30 @@ Navigate to the `Debug Contracts` tab, you should see two smart contracts displa
+</details>
+<details markdown='1'><summary>Question Two</summary>
+
+> What data structure represents the relation between keys and values (addresses to liquidity or users to ETH)??
```
two question marks

rin-st commented on README.md:
```diff
@@ -86,8 +101,38 @@ mapping (address => uint256) public liquidity;
+
+<details markdown='1'><summary>🦉 Guiding Questions</summary>
+
+<details markdown='1'><summary>Question 1</summary>
```
Question 1-4, but Question One, Two etc in other places

ayechanst replied:
fixed your 2 other comments, what is the problem with this one? i did not understand

rin-st replied:
Sorry, didn't see your answer. Just wording. In one place questions written `Question 1`, `Question 2`, ... (see above). In all other places  `Question One`, `Question Two`, ...

rin-st commented on README.md:
```diff
@@ -351,6 +688,7 @@ Let’s create two new functions that let us deposit and withdraw liquidity. How
 
 ---
 
+---
```
why to add this line?

rin-st commented:
Thanks @ayechanst and @steve0xp ! Looking great! Added some comments, please fix it and let's wait a bit for other reivews

ayechanst commented:
![individualToggles](https://github.com/scaffold-eth/se-2-challenges/assets/112447351/1720736f-5782-4f38-8950-6c2444040c04)
![oneToggle](https://github.com/scaffold-eth/se-2-challenges/assets/112447351/bb0dc8b0-1cdf-4fbd-a2db-c0808881764e)
Comparison between individual toggle and one big toggle. I thought it looked a bit awkward having all the toggles stacked up one each other. Any ideas on how we could make it look nicer? Its not a big deal whatever you guys think is appropriate is fine with me.
